src/backend/Shared/EventBus/EventBus.RabbitMQ/EventBusRabbitMQ.cs

```diff
@@ -0,0 +1,294 @@
+            DoInternalSubscription(eventName);
+
+            _subsManager.AddSubscription<T, TH>();
+            StartBasicConsume();
+        }
+
+        private void DoInternalSubscription(string eventName)
+        {
+            bool containsKey = _subsManager.HasSubscriptionsForEvent(eventName);
+            if (!containsKey)
+            {
+                if (!_persistentConnection.IsConnected)
+                {
+                    _persistentConnection.TryConnect();
+                }
+
+                using (IModel channel = _persistentConnection.CreateModel())
+                {
+                    channel.QueueBind(
+                        queue: _queueName,
+                        exchange: BROKER_NAME,
+                        routingKey: eventName);
+                }
+            }
+        }
+        #endregion
+
+        #region Unsubscribe
+        public void Unsubscribe<T, TH>()
+            where T : IntegrationEvent
+            where TH : IIntegrationEventHandler<T>
+        {
+            _subsManager.RemoveSubscription<T, TH>();
+        }
+
+        public void UnsubscribeDynamic<TH>(string eventName)
+            where TH : IDynamicIntegrationEventHandler
+        {
+            _subsManager.RemoveDynamicSubscription<TH>(eventName);
+        }
+        #endregion
+
+        #region Consume Message
+        private async Task Consumer_Received(object sender, BasicDeliverEventArgs eventArgs)
+        {
+            string eventName = eventArgs.RoutingKey;
+            string message = Encoding.UTF8.GetString(eventArgs.Body);
+
+            try
+            {
+                if (message.ToLowerInvariant().Contains("throw-fake-exception"))
+                {
+                    throw new InvalidOperationException($"Fake exception requested: \"{message}\"");
+                }
+
+                await ProcessEvent(eventName, message);
+            }
+            catch (Exception)
+            {
+                _consumerChannel.BasicNack(deliveryTag: eventArgs.DeliveryTag, multiple: false, requeue: true);
+            }
+
+            _consumerChannel.BasicAck(eventArgs.DeliveryTag, multiple: false);
+        }
+
+        private async Task ProcessEvent(string eventName, string message)
+        {
+            if (!_subsManager.HasSubscriptionsForEvent(eventName)) return;
+
+            using (ILifetimeScope scope = _autofac.BeginLifetimeScope(AUTOFAC_SCOPE_NAME))
+            {
+                IEnumerable<SubscriptionInfo> subscriptions = _subsManager.GetHandlersForEvent(eventName);
+
+                foreach (SubscriptionInfo subscription in subscriptions)
+                {
+                    object handler = scope.ResolveOptional(subscription.HandlerType);
+                    if (handler == null) return;
+
+                    if (subscription.IsDynamic)
+                    {
+                        await ProcessDynamicSubscription(handler, message);
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Add '.ConfigureAwait(false)' to this call to allow execution to continue in any thread.](https://app.codacy.com/manual/iamprovidence/Lama/pullRequest?prid=4794247)